ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -260,15 +261,16 @@ private void initiateBackgroundDeletes(List<StoreKey> idsToDelete) {
     for (StoreKey key : idsToDelete) {
       currentOperationsCount.incrementAndGet();
       currentBackgroundOperationsCount.incrementAndGet();
-      backgroundDeleter.deleteBlob(key.getID(), new FutureResult<Void>(), new Callback<Void>() {
-        @Override
-        public void onCompletion(Void result, Exception exception) {
-          if (exception != null) {
-            logger.error("Background delete operation failed with exception", exception);
-          }
-          currentBackgroundOperationsCount.decrementAndGet();
-        }
-      });
+      backgroundDeleter.deleteBlob(key.getID(), routerConfig.routerBackgroundDeleteServiceId, new FutureResult<Void>(),
```

Comment:
Would this be a little bit general? Maybe include the service this chunk blob belongs to?